src/Microsoft.FeatureManagement/FeatureManagerSnapshot.cs

```diff
@@ -16,19 +16,38 @@ namespace Microsoft.FeatureManagement
     /// </summary>
     class FeatureManagerSnapshot : IFeatureManagerSnapshot, IVariantFeatureManagerSnapshot
     {
-        private readonly IVariantFeatureManager _featureManager;
-        private readonly ConcurrentDictionary<string, ValueTask<bool>> _flagCache = new ConcurrentDictionary<string, ValueTask<bool>>();
+        private readonly IFeatureManager _featureManager;
+        private readonly IVariantFeatureManager _variantFeatureManager;
+        private readonly ConcurrentDictionary<string, Task<bool>> _flagCache = new ConcurrentDictionary<string, Task<bool>>();
```

Comment:
I don't see any reason to have two flag caches.

Reply:
Reason I did two caches is when the `IFeatureManager` and `IVariantFeatureManager` are different implementations (someone today has a custom IFeatureManager). If the `IFeatureManager` returns true for a feature while the `IVariantFeatureManager` would return false- it feels incorrect to have one take priority over the other.

We could have one cache and use a prefix or something but this felt cleaner to me.

Reply:
We aligned on a single cache to better fit the common case.